### PR TITLE
Insurance detail 30792

### PR DESCRIPTION
**Pull request by charCodelet**
## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

This is a test to see if all tests pass...no merge for now thanks.
## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/hca/config/form.js
Comment (LinnJS) [inline src/applications/hca/config/form.js:154]: Looks like all these changes are for [ticket #25411](https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/25411) and not for [ticket #30792](https://app.zenhub.com/workspaces/vsa---caregiver-5fff0cfd1462b6000e320fc7/issues/department-of-veterans-affairs/va.gov-team/30792). We should not be mixing work from more than one ticket into one PR. Makes for a lot of work, poor reviews, and hard to track code changes that are hard to revert when bugs come up. Every ticket should be a different PR if possible.

**Review by LinnJS — CHANGES_REQUESTED** [review]
Work from more than one task should not be in the same PR.

File: src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js
Comment (timwright12) [inline src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js:9]: can we remove this?

File: src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js
Comment (timwright12) [inline src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js:74]: can we remove this if it isnt being used?

File: src/applications/hca/config/chapters/veteranInformation/demographicInformation.js
Comment (timwright12) [inline src/applications/hca/config/chapters/veteranInformation/demographicInformation.js:21]: Can you use a design system class here? https://design.va.gov/utilities/color

File: src/applications/hca/config/chapters/veteranInformation/genderIdentificationInformation.js
Comment (timwright12) [inline src/applications/hca/config/chapters/veteranInformation/genderIdentificationInformation.js:11]: Can you use a design system class here? https://design.va.gov/utilities/color

File: src/applications/hca/tests/components/DemographicField.unit.spec.jsx
Comment (timwright12) [inline src/applications/hca/tests/components/DemographicField.unit.spec.jsx:10]: Can you document why these tests are being skipped?

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx
Comment (timwright12) [inline src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:1032]: These VAOS tests should be fixed in master

**Review by timwright12 — CHANGES_REQUESTED** [review]

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:1032]: Are all of the test failing?

Comment (timwright12) [reply src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:1032]: flaky test we set to skip a few days ago, a merge in from master should clear it out

File: src/applications/hca/config/chapters/insuranceInformation/general.js
Comment (Mottie) [inline src/applications/hca/config/chapters/insuranceInformation/general.js:9]: I think this will show some render errors... like a `<div>` contained within a `<p>` is invalid markup.

File: src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js
Comment (Mottie) [inline src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js:40]: same here

File: src/applications/hca/config/chapters/insuranceInformation/general.js
Comment (charCodelet) [reply src/applications/hca/config/chapters/insuranceInformation/general.js:9]: okay. thx fixed.

File: src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js
Comment (charCodelet) [reply src/applications/hca/config/chapters/insuranceInformation/providerAndPolicy.js:40]: okay. thx fixed.

File: src/applications/hca/config/chapters/veteranInformation/genderIdentificationInformation.js
Comment (LinnJS) [reply src/applications/hca/config/chapters/veteranInformation/genderIdentificationInformation.js:11]: I agree with @timwright12 need to use design system color in here on my other PR I used [vads-u-color--gray-medium](https://github.com/department-of-veterans-affairs/vets-website/pull/19337/files#diff-505884e19a88286371f1ab60c92d2ae27b4a3dd34d2d7ec6c2387b3419222430R28) here.

File: src/applications/hca/config/chapters/veteranInformation/maritalInformation.js
Comment (LinnJS) [inline src/applications/hca/config/chapters/veteranInformation/maritalInformation.js:13]: Need to revert all these changes since these changes will directly conflict with this [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/19337) and from what I can see this work does not pertain to the ticket at hand.

File: src/applications/hca/config/form.js
Comment (LinnJS) [reply src/applications/hca/config/form.js:155]: I stand by this comment and all of the following lines should be reverted to eliminate conflict with the PR stated in previous comment.

All the SIGI work does not pertain to this work, furthermore the SIGI work that is done in this PR is not behind feature toggle so will expose public to features that are not ready.

File: src/platform/forms-system/src/js/components/FormNav.jsx
Comment (LinnJS) [inline src/platform/forms-system/src/js/components/FormNav.jsx:34]: This is platform code if this code really does need to be modified needs to be discussed with VSP.

File: src/applications/hca/tests/config/general.unit.spec.jsx
Comment (LinnJS) [inline src/applications/hca/tests/config/general.unit.spec.jsx:13]: We should not be skipping test, only for debugging, need to fix test before merging so our coverage is maintained.

File: src/platform/forms-system/test/js/widgets/SSNWidget.unit.spec.jsx
Comment (LinnJS) [inline src/platform/forms-system/test/js/widgets/SSNWidget.unit.spec.jsx:15]: Same here as above.

**Review by LinnJS — CHANGES_REQUESTED** [review]
Most of my previous were not addressed, did a comprehensive review and requested additional feedback.

File: src/platform/forms-system/src/js/components/FormNav.jsx
Comment (charCodelet) [reply src/platform/forms-system/src/js/components/FormNav.jsx:34]: Okay I took that out.

File: src/platform/forms-system/test/js/widgets/SSNWidget.unit.spec.jsx
Comment (charCodelet) [reply src/platform/forms-system/test/js/widgets/SSNWidget.unit.spec.jsx:15]: This one is no longer skipped.

File: src/applications/hca/tests/config/general.unit.spec.jsx
Comment (charCodelet) [reply src/applications/hca/tests/config/general.unit.spec.jsx:13]: I have resolved this issue with a caveat that I'm still learning React Testing a bit. I figured out that general.unit needed to be broken up into general.unit AND providerAndPolicy.unit.

File: src/applications/hca/config/chapters/veteranInformation/genderIdentificationInformation.js
Comment (charCodelet) [reply src/applications/hca/config/chapters/veteranInformation/genderIdentificationInformation.js]: Design class used. Thanks.

**Review by bkjohnson — COMMENTED** [review]
> This is a test to see if all tests pass...no merge for now thanks.

For [future reference](https://depo-platform-documentation.scrollhelp.site/developer-docs/Pull-request-best-practices.688259145.html)

> If your pull request is not ready for review, please make it a draft PR. This will prevent any codeowners from receiving notifications until it is ready.

I'm going to unsubscribe from notifications - if you need a review from the Design System team once this becomes ready and it has been more than 24 hours, please submit a request in the platform support slack channel.

File: src/applications/hca/tests/config/general.unit.spec.jsx
Comment (LinnJS) [reply src/applications/hca/tests/config/general.unit.spec.jsx]: All good no worries.

**Review by vbahinwillit — COMMENTED** [review]
LGTM

**Review by timwright12 — APPROVED** [review]

**Comment by timwright12** [comment]
@charCodelet Is this still being actively worked on? If not, we should close it out to clean things up a bit

**Pull request closed without merging**
